src/app/pages/admin-panel/admin-panel.component.ts

```diff
@@ -0,0 +1,167 @@
+import { getDoc } from "@firebase/firestore"
+import { Firestore } from "@angular/fire/firestore"
+import { Component, OnInit } from "@angular/core"
+import { collection, getDocs, setDoc, doc } from "firebase/firestore"
+import {
+  Events,
+  TheBlueAllianceService,
+} from "app/services/the-blue-alliance.service"
+import { FormControl } from "@angular/forms"
+
+interface section {
```

Comment:
Interfaces should start with a capital letter.

Also, isn't Widget defined somewhere else? and schema? and section?..